### PR TITLE
HorizontalPodAutoscaler in kubectl get

### PR DESCRIPTION
**Pull request by mwielgus**
Sample output:

```
kubectl --server=127.0.0.1:8080 get horizontalpodautoscalers
NAME            REFERENCE                                       TARGET     CURRENT      MINPODS   MAXPODS   AGE
cassandra-hpa   ReplicationController/default/cassandra/scale   500m cpu   <waitning>   1         5         2h
```

### CODE REVIEW

**Comment by mwielgus** [comment]
cc: @jszczepkowski

**Comment by piosz** [comment]
cc @bgrant0607

**Comment by bgrant0607** [comment]
cc @kubernetes/kubectl

**Comment by ghodss** [comment]
Will there eventually be multiple kinds of autoscalers? If so, can we have the api be against v1/autoscalers/, and the Autoscaler object can hold a reference to AutoscalerSpec which can be a reference to a HorizontalPodAutoscaler object? Just to minimize long term pollution of the highest level object namespace, and that way all autoscalers can be treated the same at the highest level?

**Comment by mwielgus** [comment]
@ghodss 
The long term plan is to have at least 3 types of autoscalers:
* Horizontal - that will start new pods
* Vertical - that will allocate more resources to the the existing pods
* Node - that will create new kubernetes cluster nodes (provider permitting)

At this moment HorizontalPodAutoscaler is in the experimental branch of the API. Most likely it will change a bit before moving to the stable API but I (personally) doubt we will be able to have a common/generic spec and kubectl get column set for all of the autoscalers as they have different goals and config needs.

**Comment by jszczepkowski** [comment]
@ghodss 
I agree with @mwielgus. We are not planning to create a common spec for different types of autoscalers, it will not be possible.

**Comment by bgrant0607** [comment]
API groups should address pollution/scoping within the API #12951. How to handle it in kubectl is another matter, though.

**Comment by deads2k** [comment]
@liggitt vacation this week, can you unassign?

**Comment by yujuhong** [comment]
@mwielgus, please rebase.

**Comment by jlowdermilk** [comment]
`horizontalpodautoscalers` is way too long to type out in regular use. If we're adding `kubectl get` support for this resource, lets add `hpa` as a shortcut in [`expandResourceShortcut`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/kubectl.go#L93).

**Comment by jszczepkowski** [comment]
Part of #12087.

**Comment by mwielgus** [comment]
rebased, added shortcut.

File: pkg/expapi/types.go [hunk 1]
```diff
@@ -113,6 +114,10 @@ type HorizontalPodAutoscalerStatus struct {
 	// LastScaleTimestamp is the last time the HorizontalPodAutoscaler scaled the number of pods.
 	// This is used by the autoscaler to controll how often the number of pods is changed.
 	LastScaleTimestamp *util.Time `json:"lastScaleTimestamp,omitempty"`
+
+	// LastScaleTimestamp is the last time the HorizontalPodAutoscaler checked the resource consumption
```
Comment (smarterclayton) [inline pkg/expapi/types.go]: Go doc copy and paste error

**Comment by smarterclayton** [comment]
LGTM other than that one issue.

**Comment by mwielgus** [comment]
LastCheckTimestamp is not relevant for this PR anymore (HPAStatus definition changed a bit in the meantime). Removed - I will add it in a different PR.

**Comment by smarterclayton** [comment]
LGTM

**Comment by bprashanth** [comment]
@k8s-bot test this please

**Comment by brendandburns** [comment]
closing and re-opening to ping shippable...

**Comment by mwielgus** [comment]
Last commit was just a rebase.

**Comment by quinton-hoole** [comment]
e2e failure caused by github network error.  Try again.

@k8s-bot test this again please

```
Caused by: hudson.plugins.git.GitException: Command "git -c core.askpass=true fetch --tags --progress https://github.com/kubernetes/kubernetes +refs/pull/*:refs/remotes/origin/pr/*" returned status code 128:
stdout: 
stderr: error: RPC failed; result=18, HTTP code = 200
fatal: The remote end hung up unexpectedly
```

**Comment by piosz** [comment]
@k8s-bot test this please

**Pull request merged**